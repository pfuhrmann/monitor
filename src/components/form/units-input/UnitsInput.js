import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import './UnitsInput.scss';

const UnitsInput = ({ units, className, ...otherProps }) => (
  <span className={cx('Form-units-input', className)}>
    <input {...otherProps} /> {units}
  </span>
);

UnitsInput.propTypes = {
  units: PropTypes.string,
  className: PropTypes.string,
};

UnitsInput.defaultProps = {
  units: '',
  className: '',
};

export default UnitsInput;
