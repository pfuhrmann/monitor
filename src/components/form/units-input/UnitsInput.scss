@import "../../../variables.scss";

.Form-units-input {
  display: flex;
  align-items: center;
  margin-right: 1.5em;

  input {
    display: flex;
    height: 2.5em;
    padding-left: 0.5em;
    margin-right: 0.75em;
    font-size: inherit;
    color: inherit;
    outline: none;
    vertical-align: middle;
    border: 1px solid $color-gray80;
    border-radius: $border-radius;
    box-shadow: inset 0 1px 1px rgba(0, 0, 0, 0.075);
    transition: border-color ease-in-out .15s, box-shadow ease-in-out .15s;
    font-family: inherit;

    &:hover, &:focus, &:active {
      border-color: $color-mariner;
    }
  }
}
