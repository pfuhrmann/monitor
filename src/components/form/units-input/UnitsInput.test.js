import React from 'react';
import ReactDOM from 'react-dom';
import UnitsInput from './UnitsInput';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<UnitsInput type="text" id="test" onChange={() => {}} units="mmHg" />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders units input correctly', () => {
  const component = renderer.create(
    <UnitsInput type="text" id="test" onChange={() => {}} units="mmHg" />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
