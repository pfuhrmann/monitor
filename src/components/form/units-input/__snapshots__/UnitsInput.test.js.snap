// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders units input correctly 1`] = `
<span
  className="Form-units-input"
>
  <input
    id="test"
    onChange={[Function]}
    type="text"
  />
   
  mmHg
</span>
`;
